embedx/embedx.go

```diff
@@ -0,0 +1,121 @@
+package embedx
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	"github.com/tidwall/gjson"
+
+	_ "embed"
+)
+
+//go:embed config.schema.json
+var ConfigSchema string
+
+//go:embed identity_meta.schema.json
+var IdentityMetaSchema string
+
+//go:embed identity_extension.schema.json
+var IdentityExtensionSchema string
+
+type SchemaType int
+type SchemaID string
+
+const (
+	Config SchemaType = iota
+	IdentityMeta
+	IdentityExtension
+)
+
+/*var (
+	ConfigSchemaID            SchemaID
+	IdentityMetaSchemaID      SchemaID
+	IdentityExtensionSchemaID SchemaID
+)*/
+
+type Schema struct {
+	id           string
+	data         string
+	dependencies []*Schema
+}
+
+var (
+	schemas = map[SchemaType]*Schema{
+		Config: {
+			id:           gjson.Get(ConfigSchema, "$id").Str,
+			data:         ConfigSchema,
+			dependencies: nil,
+		},
+		IdentityMeta: {
+			id:   gjson.Get(IdentityMetaSchema, "$id").Str,
+			data: IdentityMetaSchema,
+			dependencies: []*Schema{
+				{
+					id:           gjson.Get(IdentityExtensionSchema, "$id").Str,
+					data:         IdentityExtensionSchema,
+					dependencies: nil,
+				},
+			},
+		},
+		IdentityExtension: {
+			id:           gjson.Get(IdentityExtensionSchema, "$id").Str,
+			data:         IdentityExtensionSchema,
+			dependencies: nil,
+		},
+	}
```

Comment:
Deduplicate? You can make this an init func as well instead, then it is only a local variable. It is equivalent for the compiler.

```suggestion
	identityExt = &Schema{
		id:           gjson.Get(IdentityExtensionSchema, "$id").Str,
		data:         IdentityExtensionSchema,
		dependencies: nil,
	}
	schemas = map[SchemaType]*Schema{
		Config: {
			id:           gjson.Get(ConfigSchema, "$id").Str,
			data:         ConfigSchema,
			dependencies: nil,
		},
		IdentityMeta: {
			id:   gjson.Get(IdentityMetaSchema, "$id").Str,
			data: IdentityMetaSchema,
			dependencies: []*Schema{identityExt},
		},
		IdentityExtension: identityExt,
	}
```